### DISCUSSION TITLE
v1.2 release

### DISCUSSION THREAD

**Discussion by txubelaxu** [txubelaxu/esphome-jk-bms#5] [Announcements]
+ NEW "Pseudo master mode". In a traditional MASTER/SLAVE configuration, during the small silences in the network ESP launches some special request to gather "device info" (frame type 03) that it is always missing in the network traffic.
+ NEW "MASTER MODE". All of the BMS are set up as slave. ESP acts as MASTER, asking periodically all the info. This mode is working.
+ More changes to HASS dashboards
+ Buffer management changed.
+ Some bugs corrected

<hr /><em>This discussion was created from the release <a href='https://github.com/txubelaxu/esphome-jk-bms/releases/tag/v1.2'>v1.2 release</a>.</em>

**Comment by Patrick197030100** [comment]
hello hi thank you for this great work a question I updated the esp32 with the file https://github.com/txubelaxu/esphome-jk-bms/blob/main/esp32-example-jkpb-rs485_1master_1slave.yaml the last one is since then I have difficulty connecting or even impossible I went back to the previous version and it works perfectly I don't understand why its bug thanks to you!

**Reply by txubelaxu** [reply]
Hi @Patrick197030100

Could you post the log info of the bug?

Many thanks!!!

**Comment by Patrick197030100** [comment]
Thank you for your quick response I am French and don't speak English! excuse me for the mistakes where to find the bug forum please I am a novice thank you

**Comment by txubelaxu** [comment]
When you launch the "run" command, what is the output printed in the screen?

**Comment by Patrick197030100** [comment]
hello look please 
![IMG_4313](https://github.com/txubelaxu/esphome-jk-bms/assets/148679768/77e9cc47-4a0e-4aff-ae52-74e6e549552e)

**Reply by txubelaxu** [reply]
Hi @Patrick197030100 ,

I think that the problem (red colored) is with your puthon3 installation.

Regards

**Comment by Patrick197030100** [comment]
python is good  not good esphone 
![IMG_4316(1)](https://github.com/txubelaxu/esphome-jk-bms/assets/148679768/28b4afaf-4b3f-489a-869d-95c6228f5138)
![IMG_4317(1)](https://github.com/txubelaxu/esphome-jk-bms/assets/148679768/266d728c-e694-4a43-b2eb-d8f3caffa488)

**Comment by Patrick197030100** [comment]
api is good

**Comment by Patrick197030100** [comment]
I always do the same thing! penultimate file works flawlessly and the last one does not recognize esp and ip address!! neither api yet it's the right one!!! I'm the only one with this problem so I put the one that works!!!

**Comment by Patrick197030100** [comment]
![IMG_4318(1)](https://github.com/txubelaxu/esphome-jk-bms/assets/148679768/3963ad11-106b-47a0-aa01-ae85abae9aba)

**Comment by Patrick197030100** [comment]
hello look please ? info bms 0 and bms 1
![Screenshot 2024-04-24 at 05-10-25 jkbms 1et 2 – Home Assistant](https://github.com/txubelaxu/esphome-jk-bms/assets/148679768/42832c22-8930-4f02-a530-81ba990c16c1)
![Screenshot 2024-04-24 at 05-10-11 jkbms 1et 2 – Home Assistant](https://github.com/txubelaxu/esphome-jk-bms/assets/148679768/01dc562d-2fcc-4beb-ae43-8395b3ef118e)

**Comment by txubelaxu** [comment]
Hi @Patrick197030100,

Your code is working %100.
It is normal not having information about 0x00 because it is not possible to asking for that information about "frame type 3rd" to 0x00 from another pseudomaster.
The slave shows the correct information.
The master shows an incomplete information because is not possible at this moment... I think JK should change the firmware to broadcast in some way that information sometimes to the RS485 network...

**Comment by jogybaer0815** [comment]
i get another 2 new JK-PB2A16S15P and one item let me not edit the setup. I have changed pw and after restart this BMS closes me out.
I think it is why i sets pw with the Windows-JK-PRG and i have pushed the reset button. I think the bms has scrumbled my password...
How can i  readout the actual password. With RS-485 i found only 1234 as "info_device_password"? Is it possible?
Any idea to solve the problem?

**Comment by Patrick197030100** [comment]
hello when I put the latest firmware for esp32 1.2 it worked and after a few days the esp32 and more recognized it was offline!!! I don't understand ! can you help me if I connect it to usb everything is ok!!!

**Reply by Patrick197030100** [reply]
![Screenshot 2024-05-04 at 08-30-27 ESPHome – Home Assistant](https://github.com/txubelaxu/esphome-jk-bms/assets/148679768/0f7e2bb8-1118-4696-afda-2e8d337069b9)

**Reply by Patrick197030100** [reply]
or the updates of Home Assistant refuses the conedction where it is the update of esphome
